dds_cli/data_putter.py

```diff
@@ -399,28 +398,15 @@ def add_file_db(self, file):
         }
 
         # Send file info to API - post if new file, put if overwrite
-        put_or_post = requests.put if fileinfo["overwrite"] else requests.post
-        try:
-            response = put_or_post(
-                DDSEndpoint.FILE_NEW,
-                params=params,
-                json=file_info,
-                headers=self.token,
-                timeout=DDSEndpoint.TIMEOUT,
-            )
-        except requests.exceptions.RequestException as err:
-            error = str(err)
-            LOG.warning(error)
-        else:
-            # Error if failed
-            if not response.ok:
-                error = f"Failed to add file '{file}' to database: {response.text}"
-                return added_to_db, error
-
-            try:
-                added_to_db, error = (True, response.json().get("message"))
-            except simplejson.JSONDecodeError as err:
-                error = str(err)
-                LOG.warning(error)
-
-        return added_to_db, error
+        request_method = "put" if fileinfo["overwrite"] else "post"
+        response_json = dds_cli.utils.perform_request(
+            DDSEndpoint.FILE_NEW,
+            method=request_method,
+            params=params,
+            json=file_info,
+            headers=self.token,
+            error_message="Failed to add file '{file}' to database",
+        )
+
+        added_to_db, message = (True, response_json)
+        return added_to_db, message
```

Comment:
In order to make sure that the upload of the rest of the files continue even if one fails we need to add a try except block here, so something like this I believe. 
```suggestion
        try:
            response_json = dds_cli.utils.perform_request(
                DDSEndpoint.FILE_NEW,
                method=request_method,
                params=params,
                json=file_info,
                headers=self.token,
                error_message="Failed to add file '{file}' to database",
            )
            added_to_db, message = (True, response_json)
        except (dds_cli.exceptions.ApiRequestError, dds_cli.exceptions.ApiResponseError, dds_cli.exceptions.DDSCLIException) as err:
            message = str(err)
            LOG.warning(message)

        return added_to_db, message
```

Reply:
I pushed the change but not sure how to simulate a file upload failure for testing right now. I continue with other tasks not to lose too much time with this testing

Reply:
Sure, that's fine, I can test and make changes in this if I see any that are needed.